### PR TITLE
pickle structured units

### LINKED ISSUE #12575: Unpickling structured quantities
### Description

Structured quantities cannot be unpickled.

### Expected behavior

If it can be serialized, it can be unserialized.

### Actual behavior

A ``TypeError`` is raised unless `protocol=0` or `1`.

### Steps to Reproduce

```python
import pickle
import astropy.units as u

m_nu = u.Quantity((0, 0, 0.6), unit="(eV, eV, eV)")
serialized = pickle.dumps(m_nu, protocol=2)
pickle.loads(serialized)
```

### System Details

macOS-10.16-x86_64-i386-64bit
Python 3.9.5 (default, May 18 2021, 12:31:01) 
[Clang 10.0.0 ]
Numpy 1.21.4
pyerfa 2.0.0.1
astropy 5.1.dev207+gbfb9252df.d20211130
Scipy 1.7.1
Matplotlib 3.3.4

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Description

Allows structured units to be pickled and unpicked when ``protocol`` >= 2.

Fixes #12575

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/units/structured.py [hunk 1]
```diff
@@ -487,6 +487,12 @@ def __ne__(self, other):
 
         return self.values() != other.values()
 
+    def __reduce_ex__(self, protocol):
```
Comment (mhvk) [inline astropy/units/structured.py:490]: I think a simpler solution is just to pass `__new__` something that will actually give the desired uninitialized `StructuredUnit`. The way you have it, the code goes through all the trouble of initializing the unit from a string, and then override the result will the old state.

I check, and the following replacement also passes all the test:
```
    def __getnewargs__(self):
        return (), None
```

Actually, reading the description of those methods at https://docs.python.org/3/library/pickle.html, it is mentioned that those are used by `copy.copy` as well - and indeed that does not work without `__getnewargs__`...  I'll make a quick commit to fix that too...
Basically this does `StructuredUnit(units=())`, which returns an empty item.

I considered just making that the default in `__new__` but then one gets a unit that cannot round-trip through `u.Unit(str(u.StructuredUnit(()))`, so seems less than ideal.

**Review by mhvk — APPROVED** [review]
@nstarman - I think with the changes I just pushed, this is all A-OK, but probably good for you to check!

File: astropy/units/structured.py (hunk 1 above)
Comment (nstarman) [reply astropy/units/structured.py]: Looks like a more elegant solution! And it's nice to pick up ``copy`` support. Thanks!

**Comment by nstarman** [comment]
Test failure in ``test_api_lookup`` is unrelated.
